### DISCUSSION TITLE
Better SAF support

### DISCUSSION THREAD

**Discussion by hj-collab** [termux/termux-packages#22155] [Ideas]
Hi,

Thank you for the app. I don't want to share complete shared storage access to Termux. I want to share a single folder. I can see Termux:API have support for SAF. But the support is not smooth. It will be nice that once termux-saf-managedir is ran then termux should be easily able to open, read and write to that particular folder. That way projects file can reside in a SAF folder without having to share full access with Termux. Current implementation is not smooth

**Comment by tareksander** [comment]
Handling SAF like regular files to other programs is effectively not possible, not without creating a proot addon and running everything that needs the folder under proot.
